### DISCUSSION TITLE
Rename macros $ref to ref

### DISCUSSION THREAD

**Discussion by bomzj** [vuejs/rfcs#474] [RFC Discussions]
### What problem does this feature solve?
Regarding https://github.com/vuejs/rfcs/pull/420
For consistency sake, rename `$ref` to `ref` in order to have the same naming convention as other macros have e.g. `defineProps`, `defineEmits`. They are not prefixed with ugly `$` sign coming from jquery/php, 'Hungarian' notations or other $$$ rich code. 

I know that $ref is experimental, just would like you to consider this once you decide to adopt an approach to reactivity transformation.

### What does the proposed API look like?

Old way
```
let x = $ref(1)
```

New way
```
let x = ref(1)
```

**Comment by leopiccionia** [comment]
Wouldn't it clash with already existing `ref`? How could it distinguish between sugared and non-sugared usage?

For example, given the following snippet:

```javascript
let x = ref(1)
useSomething(x)
```

Should the compiler translate the second line to `useSomething(x.value)` or keep it as `useSomething(x)`? Using the same identifier `ref` for both usages may avoid gradual adoption and migration.

**Comment by quanzaiyu** [comment]
本来就是为了区分`ref`才使用`$ref`作为语法糖。你的value到底是ref的value，还是响应式对象里面的value呢？
